### PR TITLE
Improve the arrival event to leg waypoint.

### PR DESCRIPTION
**Pull request by ShanMa1991**
### Description
This pr is to improve the arrival event. Right now the distance to the route leg destination to announce the arrival event is long. It would cause the arrival feedback window pops too early to cover the user location indicator and the route line. And in multi-leg route, it would also result in the arrival event of the current leg triggers too early, as far as 75 meters from the current leg destination.

### Implementation
Added the public `RouteControllerDistanceForArrival` as the distance threshold for the route leg remanning distance  of current route. When the current route leg remanning distance equal or smaller than the `RouteControllerDistanceForArrival`, the `didArriveAt` event would be announced. It would make the arrival event more precise than the `complete` route status.

### CODE REVIEW

File: Sources/MapboxCoreNavigation/CoreConstants.swift [hunk 1]
```diff
@@ -40,6 +40,13 @@ public var RouteControllerMinimumDurationRemainingForProactiveRerouting: TimeInt
  */
 public var RouteControllerProactiveReroutingInterval: TimeInterval = 120
 
+/**
+ The distance threshold to the current leg destination to anounce the arrival of the waypoint.
+ 
+ During active navigation, when the remaining distance of current route leg is equal or smaller than the `RouteControllerDistanceForArrival`, the arrival event at current leg waypoint would be announced.
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/CoreConstants.swift:46]: It is not clear that the arrival event still depends on the `.complete` status from NavNative (that depends on the route response from Directions).

Comment (bamx23) [reply Sources/MapboxCoreNavigation/CoreConstants.swift:46]: I see, that's true. Should we mention that it's not guaranteed that arrival will be triggered at this point as this variable can only decrease the remaining distance?

Comment (ShanMa1991) [reply Sources/MapboxCoreNavigation/CoreConstants.swift:46]: Thanks for the suggestion, I'll update the documentation.

**Comment by ShanMa1991** [comment]
Close this Pr because the arrival event should be triggered by the navigator side.

**Pull request closed without merging**